qualtran/surface_code/ui.py

```diff
@@ -0,0 +1,608 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any, Dict, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objects as go
+from dash import ALL, Dash, dcc, html, Input, Output
+from dash.exceptions import PreventUpdate
+
+from qualtran.surface_code import ccz2t_cost_model, fifteen_to_one, magic_state_factory
+from qualtran.surface_code import quantum_error_correction_scheme_summary as qecs
+from qualtran.surface_code import rotation_cost_model
+from qualtran.surface_code.algorithm_summary import AlgorithmSummary
+from qualtran.surface_code.azure_cost_model import code_distance, logical_qubits, minimum_time_steps
+from qualtran.surface_code.ccz2t_cost_model import (
+    get_ccz2t_costs_from_grid_search,
+    iter_ccz2t_factories,
+)
+from qualtran.surface_code.magic_count import MagicCount
+from qualtran.surface_code.multi_factory import MultiFactory
+
+
+def get_objects(modules, object_type):
+    for module in modules:
+        for obj_name in dir(module):
+            obj = getattr(module, obj_name)
+            if isinstance(obj, object_type):
+                yield obj_name, obj
+
+
+QEC_SCHEMES = dict(get_objects([qecs], qecs.QuantumErrorCorrectionSchemeSummary))
+MAGIC_FACTORIES = dict(
+    get_objects([fifteen_to_one, ccz2t_cost_model], magic_state_factory.MagicStateFactory)
+)
+ROTATION_MODELS = dict(get_objects([rotation_cost_model], rotation_cost_model.RotationCostModel))
+SUPPORTED_ESTIMATION_MODELS = [
+    'GidneyFolwer (arxiv:1812.01238)',
+    'Beverland et al (arxiv:2211.07629)',
```

Comment:
Can we make these enums / constants instead of strings? Comparing against the exact string at multiple places in the code seems very error prone.

Reply:
not an enum. but constants should do the trick here.